mibidata/tiff.py

```diff
@@ -289,29 +275,39 @@ def _page_metadata(page, description):
     date = datetime.datetime.strptime(
         page.tags['datetime'].value.decode(ENCODING),
         _DATETIME_FORMAT)
-    return {
-        'run': description.get('mibi.run'),
+
+    # check version for backwards compatibility
+    _convert_from_previous_metadata_versions(description)
+
+    metadata = {key.split('.')[1]: value for key, value in \
+                description.items() if key.startswith('mibi.')}
+    metadata.update({
         'coordinates': (
             _cm_to_micron(page.tags['x_position'].value),
             _cm_to_micron(page.tags['y_position'].value)),
         'date': date,
-        'size': size,
-        'slide': description.get('mibi.slide'),
-        'point_name': description.get('mibi.description'),
-        'folder': description.get('mibi.folder'),
-        'dwell': description.get('mibi.dwell'),
-        'scans': description.get('mibi.scans'),
-        'aperture': description.get('mibi.aperture'),
-        'instrument': description.get('mibi.instrument'),
-        'tissue': description.get('mibi.tissue'),
-        'panel': description.get('mibi.panel'),
-        'mass_offset': description.get('mibi.mass_offset'),
-        'mass_gain': description.get('mibi.mass_gain'),
-        'time_resolution': description.get('mibi.time_resolution'),
-        'miscalibrated': description.get('mibi.miscalibrated'),
-        'check_reg': description.get('mibi.check_reg'),
-        'filename': description.get('mibi.filename'),
-    }
+        'size': size})
+
+    return metadata
+
+
+def _convert_from_previous_metadata_versions(description):
+    """Convert old metadata format to new one.
+
+    This function ensures backwards compatibility for previous versions.
+    """
+    try:
+        description['mibi.version']
+    except KeyError:
+        # if the key doesn't exist it means that the version was None and hence
+        # not saved in the tiff file
+        description['mibi.version'] = None
+
+    if description['mibi.version'] != mi.MIBITIFF_VERSION:
+        description['mibi.fov_name'] = description['mibi.description']
```

Comment:
Again here I think it's better to check on whether ``fov_name'` is present rather than relying on the version.

Reply:
@rfinck I think in this case we should rely on the version, and assume the user is familiar with the required keywords of the current version. Otherwise, after a few versions, it will be very hard to follow the logic of this function, with multiple if and try checks on different variables.
I implemented unit tests to ensure backwards compatibility. What we would need at some point is a better way to compare MIBItiff versions, but this can be done in a separate PR. Let me please know if you still want me to change this.